### DISCUSSION TITLE
Failure when starting fresh docker server

### DISCUSSION THREAD

**Discussion by malake** [mbround18/valheim-docker#640] [Help!]
Hello, I'm trying to stand up a new server by running docker but can't get the image to start. When running a fresh server with all the current defaults (except DEBUG_MODE =1) this is the current log output:
```
[Valheim][root] : ###########################################################################
[Valheim][root] : Valheim Server - Sun 18 Dec 2022 03:05:59 PM EST
[Valheim][root] : Initializing your container...
[Valheim][root] : You must be in development. Good luck!
[Valheim][root] : ###########################################################################
[Valheim][root] : Switching UID and GID
[Valheim][root] : usermod: no changes
[Valheim][root] :
Configuring Preset Env
Preset Env Configured
[Valheim][root] : Setting up file systems
[Valheim][root] : Navigating to steam home...
[Valheim][root] : Launching as steam...
[Valheim][steam]: ###########################################################################
[Valheim][steam]: Valheim Server - Sun 18 Dec 2022 03:06:00 PM EST
[Valheim][steam]: STEAM_UID 1000 - STEAM_GUID 1000
[Valheim][steam]: Installing Valheim via odin 1.5.3...
[Valheim][steam]: ###########################################################################
[Valheim][steam]: Variables loaded.....
[Valheim][steam]: Port: 2456
[Valheim][steam]: Name: Valheim Docker
[Valheim][steam]: World: Dedicated
[Valheim][steam]: Public: 1
[Valheim][steam]: With Crossplay:
[Valheim][steam]: Password: (REDACTED)
[Valheim][steam]: Running Install...
[ODIN][DEBUG] - Debugging set to true
[ODIN][DEBUG] - Debug mode enabled!
[ODIN][DEBUG] - Launching install command...
[ODIN][DEBUG] - Env var default 'ODIN_WORKING_DIR': '/home/steam/valheim'
[ODIN][INFO]  - Installing 896660 to /home/steam/valheim
[ODIN][DEBUG] - Env var default 'ODIN_WORKING_DIR': '/home/steam/valheim'
[ODIN][INFO]  - steamcmd found in path
[ODIN][DEBUG] - Env var found 'DEBUG_MODE': '1'
[ODIN][DEBUG] - Env var default 'USE_PUBLIC_BETA': '0'
[ODIN][DEBUG] - Launching install command: "/usr/bin/steamcmd" "+force_install_dir /home/steam/valheim" "+login anonymous" "verbose" "+app_update 896660" "+quit"
Redirecting stderr to '/home/steam/.steam/logs/stderr.txt'
Looks like steam didn't shutdown cleanly, scheduling immediate update check
threadtools.cpp (3787) : Assertion Failed: Probably deadlock or failure waiting for thread to initialize.
Thread failed to initialize
ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
[  0%] Checking for available update...
threadtools.cpp (3787) : Assertion Failed: Probably deadlock or failure waiting for thread to initialize.
Thread failed to initialize
CWorkThreadPool::StartWorkThread: Thread creation failed.
Exiting on SPEW_ABORT
[ODIN][ERROR] - Exited with http code: 1
```

I'm running this on a Synology DSM using docker. I previously ran lloesche/lloesche-valheim-server fine until I started trying to update for Mistlands . Your project appears to be better maintained and significantly more active :-)

**Comment by mbround18** [comment]
Ooo that's a steam error, maybe it doesn't have permissions? Try blowing away the server folder

**Comment by spannerman79** [comment]
I'd say its permissions issue;

From the log;

```
[Valheim][steam]: STEAM_UID 1000 - STEAM_GUID 1000
```

I'd say UID & GID for the host directory aren't `1000:1000` and need to be set correctly - see https://github.com/mbround18/valheim-docker#container-env-variables for vars `PUID` & `PGID` and set accordingly.

In the mean time nuke the host mount mount and set the correct `UID/GID` as what @mbround18 stated.

**Comment by malake** [comment]
Thanks, y'all. I appreciate you responding 😁 I'm traveling for the
holidays at the moment and will check it out when I get back home.

I plan on writing up the documentation on deploying to a Synology and
opening a PR once this gets working.

**Reply by mbround18** [reply]
@malake  any updates?

**Reply by malake** [reply]
@mbround18, I'd destroyed the entire setup and recreated to no luck. I haven't looked at it since and all of my friends have abandoned Valheim for other endeavors. Although in the spirit of OSS, I'll take a look at this when I get a free evening and try again.

**Comment by lollek** [comment]
Hey @malake . I just had the same issue - had a long running server which I've played on just a few months ago and the exact same error when restarting the server to update it for Mistlands. 

In my case, it turned out that the host OS running docker needed to be updated, like in `apt get dist-upgrade`. I don't know if it's something specific Steam checks for, or if some libs were old, like libpthreads. 

Not very helpful, I know, but it can hopefully nudge you in the right direction.

**Reply by malake** [reply]
What's your OS? Are you running on a VM locally, physical machine, or a Synology?

**Reply by lollek** [reply]
My server is a KVM running Debian. It used to run Debian 9 and I updated it to Debian 11, which makes it a large leap. E.g. I think kernel version changed from 4.x to 5.10. 

You could try running the [base steamcmd image](https://github.com/steamcmd/docker) directly - Something like `docker run --rm -it steamcmd/steamcmd:ubuntu +login anonymous +app_update 896660 +quit` and see if that fails as well.
